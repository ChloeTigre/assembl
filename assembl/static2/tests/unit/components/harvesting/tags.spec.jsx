import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';

import Tags from '../../../../js/app/components/harvesting/tags';

const tags = [{ label: 'foo', value: 'fooid' }, { label: 'bar', value: 'barid' }];

describe('Tags component', () => {
  const updateTags = jest.fn(() => {});

  it('should match Tags snapshot', () => {
    const props = {
      initialValues: tags,
      updateTags: updateTags,
      canEdit: true
    };
    const shallowRenderer = new ShallowRenderer();
    shallowRenderer.render(<Tags {...props} />);
    const result = shallowRenderer.getRenderOutput();
    expect(result).toMatchSnapshot();
  });

  it('should match Tags snapshot (user can\'t edit)', () => {
    const props = {
      initialValues: tags,
      updateTags: updateTags,
      canEdit: false
    };
    const shallowRenderer = new ShallowRenderer();
    shallowRenderer.render(<Tags {...props} />);
    const result = shallowRenderer.getRenderOutput();
    expect(result).toMatchSnapshot();
  });

  it('should match Tags snapshot without initialValues', () => {
    const props = {
      initialValues: [],
      updateTags: updateTags,
      canEdit: true
    };
    const shallowRenderer = new ShallowRenderer();
    shallowRenderer.render(<Tags {...props} />);
    const result = shallowRenderer.getRenderOutput();
    expect(result).toMatchSnapshot();
  });
});