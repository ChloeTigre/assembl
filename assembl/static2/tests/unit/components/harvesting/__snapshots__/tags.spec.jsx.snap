// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tags component should match Tags snapshot (user can't edit) 1`] = `
<div
  className="harvesting-tags-container"
>
  <label
    htmlFor="tags"
  >
    Tags
  </label>
  <div
    className="harvesting-tags-titles-container"
  >
    <Apollo(Tag)
      canRemove={false}
      excludeOptions={
        Array [
          "fooid",
          "barid",
        ]
      }
      onUpdate={[Function]}
      remove={[Function]}
      tag={
        Object {
          "label": "foo",
          "value": "fooid",
        }
      }
    />
    <Apollo(Tag)
      canRemove={false}
      excludeOptions={
        Array [
          "fooid",
          "barid",
        ]
      }
      onUpdate={[Function]}
      remove={[Function]}
      tag={
        Object {
          "label": "bar",
          "value": "barid",
        }
      }
    />
  </div>
</div>
`;

exports[`Tags component should match Tags snapshot 1`] = `
<div
  className="harvesting-tags-container"
>
  <label
    htmlFor="tags"
  >
    Tags
  </label>
  <div
    className="harvesting-tags-titles-container"
  >
    <Apollo(Tag)
      canRemove={true}
      excludeOptions={
        Array [
          "fooid",
          "barid",
        ]
      }
      onUpdate={[Function]}
      remove={[Function]}
      tag={
        Object {
          "label": "foo",
          "value": "fooid",
        }
      }
    />
    <Apollo(Tag)
      canRemove={true}
      excludeOptions={
        Array [
          "fooid",
          "barid",
        ]
      }
      onUpdate={[Function]}
      remove={[Function]}
      tag={
        Object {
          "label": "bar",
          "value": "barid",
        }
      }
    />
    <div
      className="harvesting-tags-edit"
      onClick={[Function]}
    >
      +
    </div>
  </div>
</div>
`;

exports[`Tags component should match Tags snapshot without initialValues 1`] = `
<div
  className="harvesting-tags-container"
>
  <label
    htmlFor="tags"
  >
    Tags
  </label>
  <div
    className="harvesting-tags-titles-container"
  >
    <div
      className="harvesting-tags-edit"
      onClick={[Function]}
    >
      +
    </div>
  </div>
</div>
`;
